omgeo/services/__init__.py

```diff
@@ -658,9 +658,11 @@ def _geocode(self, pq):
                 query = dict(query, searchExtent=pq.viewbox.to_esri_wgs_json())
         else: # single-line
             method = 'find'
+            magic_key = pq.key if hasattr(pq, 'key') else ''
```

Comment:
I'm inclined to leave it to the user. I updated the docs with notes. You kind of have to be looking for it anyway since it's not a named parameter on the PlaceQuery so it will likely only get used by advanced users or those inclined to read the code. And if you are passing in text that doesn't match the key along with a bounding box that doesn't cover the location, you might have bigger problems.